### ISSUE TITLE
[fxa upgrade ] crashes when accessing `Manage Account & Devices` without internet connection

### ISSUE THREAD

**Issue by abodea** [mozilla-mobile/firefox-ios#6268]
## Steps to reproduce
Launch FF
Open the hamburger menu
Turn off the internet connection
Tap on the sync button few times
Tap on the account(where the name is displayed)
Access Manage Account & Devices 

### Expected behavior
An error that there is no internet connection or nothing at all should happen as the menu cannot be accessed.
### Actual behavior
Firefox gets stuck after accessing `Manage Account & Devices`. and when restarting the app, the crash prompt is displayed.

### Device information
* Device: **iPhone 8 (13.2.3)**
* Version: **Latest Beta 23.1(17435)**

### Notes
[video](https://youtu.be/5zsk13nChHc)
[JetsamEvent-2020-03-13-113951.zip](https://github.com/mozilla-mobile/firefox-ios/files/4329982/JetsamEvent-2020-03-13-113951.zip)

**Comment by garvankeeley** [comment]
```
#0	0x00000001a9d59c94 in __psynch_cvwait ()
#1	0x00000001a9c7b094 in _pthread_cond_wait$VARIANT$armv81 ()
#2	0x0000000107be6664 in std::sys::unix::condvar::Condvar::wait_timeout::h4bba326e57cd757a at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/sys/unix/condvar.rs:157
#3	0x0000000107bc8920 in std::sys_common::condvar::Condvar::wait_timeout::h915d4bb45ed1b4f1 [inlined] at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/sys_common/condvar.rs:51
#4	0x0000000107bc8914 in std::sync::condvar::Condvar::wait_timeout::h13fb47d2ee14964a [inlined] at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/sync/condvar.rs:405
#5	0x0000000107bc88fc in std::thread::park_timeout::h37f888eb9c416536 at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/thread/mod.rs:1006
#6	0x000000010734a290 in reqwest::blocking::wait::timeout::h4d0d0949f7a24fda at /Users/gkeeley/.cargo/registry/src/github.com-1ecc6299db9ec823/reqwest-0.10.4/src/blocking/wait.rs:44
#7	0x00000001073a983c in reqwest::blocking::client::ClientHandle::execute_request::h2643f141e92c61e5 at /Users/gkeeley/.cargo/registry/src/github.com-1ecc6299db9ec823/reqwest-0.10.4/src/blocking/client.rs:767
#8	0x00000001073a712c in reqwest::blocking::client::Client::execute::h953a72aa5517e972 at /Users/gkeeley/.cargo/registry/src/github.com-1ecc6299db9ec823/reqwest-0.10.4/src/blocking/client.rs:623
#9	0x0000000107254190 in viaduct::backend::reqwest::send::h8607be3be3dfc83b at /Users/gkeeley/code/application-services/components/viaduct/src/backend/reqwest.rs:69
#10	0x000000010725167c in viaduct::backend::send::h822e6f1eeb2fb4cb at /Users/gkeeley/code/application-services/components/viaduct/src/backend.rs:46
#11	0x0000000107233204 in viaduct::Request::send::ha75414798842ef47 at /Users/gkeeley/code/application-services/components/viaduct/src/lib.rs:83
#12	0x0000000106f837f4 in fxa_client::http_client::Client::make_request::h01defb9080057102 at /Users/gkeeley/code/application-services/components/fxa-client/src/http_client.rs:385
#13	0x0000000106f80c88 in _$LT$fxa_client..http_client..Client$u20$as$u20$fxa_client..http_client..FxAClient$GT$::devices::h2d1258cb0e935b2e at /Users/gkeeley/code/application-services/components/fxa-client/src/http_client.rs:314
#14	0x00000001070bbc24 in fxa_client::device::_$LT$impl$u20$fxa_client..FirefoxAccount$GT$::get_devices::h0d4aa988ee194444 at /Users/gkeeley/code/application-services/components/fxa-client/src/device.rs:25
#15	0x0000000106effcac in fxaclient_ffi::fxa_get_devices::_$u7b$$u7b$closure$u7d$$u7d$::h5ad021f1eafaf9c6 at /Users/gkeeley/code/application-services/components/fxa-client/ffi/src/lib.rs:399
#16	0x0000000106f33cac in ffi_support::handle_map::ConcurrentHandleMap$LT$T$GT$::call_with_result_mut::_$u7b$$u7b$closure$u7d$$u7d$::h10e25cf6767f02b9 at /Users/gkeeley/code/application-services/components/support/ffi/src/handle_map.rs:948
#17	0x0000000106f197ac in ffi_support::call_with_result_impl::_$u7b$$u7b$closure$u7d$$u7d$::h06d7db58751dc9b7 at /Users/gkeeley/code/application-services/components/support/ffi/src/lib.rs:226
#18	0x0000000106f25880 in std::panicking::try::do_call::h1d04527f6862e6e7 at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/panicking.rs:292
#19	0x0000000107befa1c in __rust_maybe_catch_panic at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libpanic_unwind/lib.rs:80
#20	0x0000000106f22fa8 in std::panicking::try::h5c8b90de7a12c3a7 at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/panicking.rs:271
#21	0x0000000106f13bf8 in std::panic::catch_unwind::hc0cd722103a7596d at /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libstd/panic.rs:394
#22	0x0000000106f192e0 in ffi_support::call_with_result_impl::hf73570b690981fa0 at /Users/gkeeley/code/application-services/components/support/ffi/src/lib.rs:224
#23	0x0000000106f14dfc in ffi_support::call_with_result::h7c6aa18553ecf18d at /Users/gkeeley/code/application-services/components/support/ffi/src/lib.rs:192
#24	0x0000000106f3212c in ffi_support::handle_map::ConcurrentHandleMap$LT$T$GT$::call_with_result_mut::h726f80f177dc5d09 at /Users/gkeeley/code/application-services/components/support/ffi/src/handle_map.rs:941
#25	0x0000000106f21364 in fxa_get_devices at /Users/gkeeley/code/application-services/components/fxa-client/ffi/src/lib.rs:398
#26	0x0000000106a3e37c in closure #1 in RustFxAccount.fetchDevices() at /Users/gkeeley/code/application-services/components/fxa-client/ios/FxAClient/RustFxAccount.swift:178
#27	0x0000000106a3b92c in thunk for @callee_guaranteed (@unowned UnsafeMutablePointer<FxAError>) -> (@unowned FxARustBuffer?, @error @owned Error) ()
#28	0x0000000106a41848 in thunk for @callee_guaranteed (@unowned UnsafeMutablePointer<FxAError>) -> (@unowned FxARustBuffer?, @error @owned Error)partial apply ()
#29	0x0000000106a41c1c in closure #1 in closure #1 in rustCall<A>(_:) at /Users/gkeeley/code/application-services/components/fxa-client/ios/FxAClient/RustFxAccount.swift:299
#30	0x0000000106a42258 in partial apply for closure #1 in closure #1 in rustCall<A>(_:) ()
#31	0x00000001df6ba4bc in partial apply for thunk for @callee_guaranteed () -> (@out A, @error @owned Error) ()
#32	0x00000001df6b1a28 in closure #1 in closure #1 in OS_dispatch_queue._syncHelper<A>(fn:execute:rescue:) ()
#33	0x00000001df6ba2b0 in partial apply for thunk for @callee_guaranteed () -> () ()
#34	0x00000001df6ac208 in thunk for @escaping @callee_guaranteed () -> () ()
#35	0x000000010b12abd8 in _dispatch_client_callout ()
#36	0x000000010b139858 in _dispatch_lane_barrier_sync_invoke_and_complete ()
#37	0x00000001df6b1fdc in @nonobjc OS_dispatch_queue.sync(execute:) ()
#38	0x00000001df6b780c in partial apply for @nonobjc  ()
#39	0x00000001df6b18b4 in OS_dispatch_queue._syncHelper<A>(fn:execute:rescue:) ()
#40	0x00000001df6b1f1c in OS_dispatch_queue.sync<A>(execute:) ()
#41	0x0000000106a410bc in closure #1 in rustCall<A>(_:) at /Users/gkeeley/code/application-services/components/fxa-client/ios/FxAClient/RustFxAccount.swift:298
#42	0x0000000106a41154 in partial apply for closure #1 in rustCall<A>(_:) ()
#43	0x0000000106a4b9d0 in static FirefoxAccountError.tryUnwrap<A>(_:) at /Users/gkeeley/code/application-services/components/fxa-client/ios/FxAClient/Errors/FxAError.swift:68
#44	0x0000000106a4b7d0 in static FirefoxAccountError.unwrap<A>(_:) at /Users/gkeeley/code/application-services/components/fxa-client/ios/FxAClient/Errors/FxAError.swift:59
#45	0x0000000106a3ae20 in rustCall<A>(_:) at /Users/gkeeley/code/application-services/components/fxa-client/ios/FxAClient/RustFxAccount.swift:297
#46	0x0000000106a3e148 in RustFxAccount.fetchDevices() at /Users/gkeeley/code/application-services/components/fxa-client/ios/FxAClient/RustFxAccount.swift:177
#47	0x0000000106a375dc in closure #1 in FxAccount.fetchDevices() at /Users/gkeeley/code/application-services/components/fxa-client/ios/FxAClient/FxAccount.swift:72
#48	0x0000000106a37678 in partial apply for closure #1 in FxAccount.fetchDevices() ()
#49	0x0000000106a376a0 in thunk for @callee_guaranteed () -> (@owned [Device], @error @owned Error) ()
#50	0x0000000106a3773c in partial apply for thunk for @callee_guaranteed () -> (@owned [Device], @error @owned Error) ()
#51	0x0000000106a39e64 in FxAccount.notifyAuthErrors<A>(_:) at /Users/gkeeley/code/application-services/components/fxa-client/ios/FxAClient/FxAccount.swift:191
#52	0x0000000106a37524 in FxAccount.fetchDevices() at /Users/gkeeley/code/application-services/components/fxa-client/ios/FxAClient/FxAccount.swift:71
#53	0x0000000106a436c8 in closure #1 in DeviceConstellation.refreshState() at /Users/gkeeley/code/application-services/components/fxa-client/ios/FxAClient/FxAccountDeviceConstellation.swift:37
```

**Comment by garvankeeley** [comment]
Filed to a-s https://github.com/mozilla/application-services/issues/2811

**Comment by garvankeeley** [comment]
Fixed by 0dca8bad912b12a20bdf8a56a55ccd4073f84719, new beta with the fix in a few hours

**Comment by abodea** [comment]
Verified as fixed on the latest beta build **23.1(17476)** with **iPhone X (13.3.1).** 
Note that I tried more than 30 times but I wasn't able to reproduce the crash.
I also added 2 new issues regarding this, maybe they are related: https://github.com/mozilla-mobile/firefox-ios/issues/6285 https://github.com/mozilla-mobile/firefox-ios/issues/6286
